### PR TITLE
Use record gear to save profile

### PR DESCRIPTION
**Pull request by kaojohnny**
connects #755

### CODE REVIEW

File: pkg/auth/dependency/userprofile/recordstore.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package userprofile
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/franela/goreq"
```
Comment (Steven-Chan) [inline pkg/auth/dependency/userprofile/recordstore.go]: why still need `goreq` 🤔

**Comment by Steven-Chan** [comment]
@kaojohnny is this PR ready to review? I find the `getUserProfile` is calling record gear api instead of importing record code.

Other than this I would merge the PR. I would open an issue about refactoring the record store. Because the UserProfileStore is now importing code from handler, which looks very bad.

**Comment by Steven-Chan** [comment]
@rickmak @carmenlau any comment?

**Comment by kaojohnny** [comment]
@Steven-Chan I made a update of using record gear API to query user profile record. Please help review. Thanks!

**Comment by rickmak** [comment]
Can someone write down the discussion point? Seem @Steven-Chan don't provide any argument here but @kaojohnny does.

**Comment by kaojohnny** [comment]
@rickmak After some consideration, I think that use two different invoking styles (importing record gear API and go through public API) may confuse other developers, so I decide to import record gear API to both endpoints to avoid confusion.

**Pull request merged**